### PR TITLE
feat(datastore): add error maps for error handler

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

Add additional errors maps for error handler

#### Issue #, if available

N/A

#### Description of how you validated changes

Tested each scenario.

* Server errors will have a 5xx response
* Timeout
* Newly required field with outdated client will give error like `Variable 'field' has coerced Null value for NonNull type`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/src/sync/processors/errorMaps.ts [hunk 1]
```diff
@@ -4,35 +4,59 @@ export type ErrorMap = Partial<{
...
 export const syncErrorMap: ErrorMap = {
+	BadModel: () => false,
 	BadRecord: error => /^Cannot return \w+ for [\w-_]+ type/.test(error.message),
 	ConfigError: () => false,
-	Transient: () => false,
+	Transient: error => connectionTimeout(error) || serverError(error),
 	Unauthorized: () => false,
 };
 
+function unwrapObserbableError(observableError: any) {
```
Comment (svidgen) [inline packages/datastore/src/sync/processors/errorMaps.ts:50]: Let's start adding good docstrings more obsessively, please. I want us to be able to see a helpful description of a method by hovering over its name wherever it's being called. I realize this one is pretty local _for now_. But, any time we add a function, class, etc.. -- *or find ourselves having to F12 to see what one's doing* -- let's get a good enough docstring in place to never have to hit F12 again to understand what's going on in the caller code.

**Review by svidgen — COMMENTED** [review]
Left a comment about docstrings. Can we also please circle back on this with some tests?

File: packages/datastore/__tests__/mutation.test.ts [hunk 2]
```diff
@@ -22,6 +23,7 @@ let modelInstanceCreator: any;
 let Model: PersistentModelConstructor<ModelType>;
 let PostCustomPK: PersistentModelConstructor<PostCustomPKType>;
 let PostCustomPKSort: PersistentModelConstructor<PostCustomPKSortType>;
+let axiosError;
```
Comment (svidgen) [inline packages/datastore/__tests__/mutation.test.ts:26]: Non-blocking, but could use a comment here explaining how this is used / where this is thrown &mdash; or even just that fact that a Rest mock is throwing whatever error is present here.

**Review by svidgen — APPROVED** [review]

File: packages/datastore/src/sync/processors/errorMaps.ts (hunk 1 above)
Comment (svidgen) [reply packages/datastore/src/sync/processors/errorMaps.ts]: Thanks!

**Comment by svidgen** [comment]
Thanks for the tests! 🥳

**Review by david-mcafee — APPROVED** [review]
LGTM! 🎉

**Pull request merged**